gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -210,13 +253,11 @@ static __always_inline void trace_v4(struct pt_regs *ctx, __u64 pid_tgid,
 	event->tid = (u32)pid_tgid;
 	event->uid = (u32)uid_gid;
 	event->gid = (u32)(uid_gid >> 32);
-	event->src.version = event->dst.version = 4;
-	event->src.proto = event->dst.proto = IPPROTO_TCP;
-	BPF_CORE_READ_INTO(&event->src.addr_raw.v4, sk,
-			   __sk_common.skc_rcv_saddr);
-	BPF_CORE_READ_INTO(&event->dst.addr_raw.v4, sk, __sk_common.skc_daddr);
-	event->dst.port = dport;
-	event->src.port = BPF_CORE_READ(sk, __sk_common.skc_num);
+	if (src_dst_entry) {
+		event->src = src_dst_entry->src;
+		event->dst = src_dst_entry->dst;
+	} else
+		read_l4endpoints_from_sock_v4(&event->src, &event->dst, sk);
```

Comment:
It can happen that src/dst is already there. For example when calling `connect` on an already connected socket. I don't know for `SS_CONNECTING` though.

In general any other path/case than https://elixir.bootlin.com/linux/v6.11/source/net/ipv4/af_inet.c#L654

Reply:
You're right. Could you please add a comment to clarify it? Something like:

```suggestion
	} else {
		// src_dst_entry is not set when tcp_v{4|6}_connect is not called, e.g.,
		// when calling inet_stream_connect on an already connected socket. So,
		// try to read the endpoints from the socket here.
		read_l4endpoints_from_sock_v4(&event->src, &event->dst, sk);
	}
```

Same in `trace_v6` and `exit_inet_stream_connect`.